### DISCUSSION TITLE
404 for other routes except index

### DISCUSSION THREAD

**Discussion by eokwukwe** [leokhoa/laragon#454] [Q&A] [answered]
I am trying to create new php only project using Laragon. My pretty URL is working `demo.test` and it loads the home page of the project. I am getting 404 error when I try to use `demo.test/contact` or `demo.test/about` page. Browser hits `index.php` only for the `first.test` for other two cases I am getting 404. 

Anyone with idea on how to deal with this issue?

**Comment by LebToki** [comment, marked as answer]
It sounds like you need to set up URL rewriting in your project. This will allow your web server to redirect requests for specific URLs to the appropriate PHP file.

Here's how you can set up URL rewriting using Apache and a .htaccess file:

Create a new file named '.htaccess' in the root directory of your project.
Add the following code to the file:
```

RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^(.*)$ index.php/$1 [L]

```
Save the file and try accessing your 
demo.test/contact
 or 
demo.test/about

 pages again.

This code tells Apache to enable URL rewriting, and to redirect all requests that are not for existing files or directories to 
index.php , with the original URL appended as a parameter.

**Reply by eokwukwe** [reply]
Thank you @LebToki.

**Reply by R0ck3n** [reply]
Thank you @LebToki. Same problem and you solve it !

**Reply by AKMimi** [reply]
Thank you @LebToki your answer worked for me. I had the same problem.

**Comment by eokwukwe** [comment]
@LebToki I want to update the app structure so that the root document is in the public folder. How do I achieve that so that when I still visit `demo.test` it will route to the `index.php` file in the public folder? I don't want to change the global document root from the Laragon setting but just for this project.

**Reply by LebToki** [reply]
If you are feddling with lots of Laravel projects. I recommend modifying line 489 where the Laravel switch case is located, and changing it line 494 to point to /Public/index.php. This will result in the following code:

` // LARAVEL
        case file_exists($host.'/public/index.php') && is_dir($host.'/app') && file_exists($host.'/.env'):
            $app_name = ' Laravel ';
            $avatar = 'assets/Laravel.png';
            $admin_link = '
                                <a href="'.$url.'://'.$host.'.local/Public" target="_blank">
                                    <small style="font-size: 8px; color: #cccccc;">
                                        '.$app_name.'
                                    </small>
                                    <br>
                                    Public Folder
                                </a>';
            break;`

**Reply by eokwukwe** [reply]
Okay. This project is not a Laravel project (Laravel projects work automatically without any additional configuration). This is just a simple PHP project I am using to refresh the basics of PHP. 

However, I was able to achieve what I want by editing the ROOT in `auto.demo.test.conf` file in the `Apache sites-enabled` settings to `C:/laragon/www/demo/public` instead of `C:/laragon/www/demo`.

Thank you for your help.

**Reply by LebToki** [reply]
anytime my friend was about to propose a redirect too
enjoy coding!

**Reply by ysolis-nube** [reply]
Hi, I've set everything up correctly so far, but I'm no longer getting a 404 error, but rather a "No input filespecific" error. What could be the cause?
